### PR TITLE
[flink] partial-update merge engine support batch deletion using flink sql

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

partial-update merge engine support batch deletion using flink sql

### Tests

paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/PartialUpdateITCase#testRemoveRecordOnDelete

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SupportsRowLevelOperationFlinkTableSink.java [hunk 1]
```diff
@@ -185,11 +185,18 @@ private void validateDeletable() {
                             table.getClass().getName()));
         }
 
-        MergeEngine mergeEngine = CoreOptions.fromMap(table.options()).mergeEngine();
-        if (mergeEngine != DEDUPLICATE) {
-            throw new UnsupportedOperationException(
-                    String.format("Merge engine %s can not support batch delete.", mergeEngine));
+        CoreOptions coreOptions = CoreOptions.fromMap(table.options());
+        if (coreOptions.mergeEngine() == DEDUPLICATE
+                || coreOptions.ignoreDelete()
```
Comment (Aitozi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SupportsRowLevelOperationFlinkTableSink.java:190]: If its `ignoreDelete`, do we have to allow it execute `delete from`?

Comment (zhuangchong) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SupportsRowLevelOperationFlinkTableSink.java]: Yes, If its ignoreDelete, it should not be allowed to execute `delete from`.

**Review by Aitozi — APPROVED** [review]
LGTM

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**